tests/functional/triggering/00-recovery/flow.cylc

```diff
@@ -17,10 +17,10 @@
     final cycle point = 20110101T12
     [[graph]]
         T00,T12 = """
-           pre:finish => model  # finish trigger
-           model:fail => diagnose => recover # fail trigger
-           model => !diagnose & !recover  # explicit success trigger
-           model:succeed | recover => post  # conditional and explicit success
+           pre:finish => model?  # finish trigger
+           model:fail? => diagnose => recover # fail trigger
+           model? => !diagnose & !recover  # explicit success trigger
+           model:succeed? | recover => post  # conditional and explicit success
```

Comment:
I think this comes under "explicit is better than implicit" -- PEP20.

Allowing users to optionally omit the optional output could get more confusing still.

Reply:
Yeah we discussed this. The idea was that requiring consistency wherever the same task output appears is safer. Otherwise you can't be sure that an output is optional or required without mentally parsing the whole graph.